src/util/reader.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) Contributors to the Suwayomi project
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { IChapter } from 'typings';
+import client from 'util/client';
+import makeToast from 'components/util/Toast';
+
+export const shouldSkipChapter = async (chapterIndex: number, currentChapter: IChapter) => {
+    const nextChapter = (await client.get<IChapter>(`/api/v1/manga/${currentChapter.mangaId}/chapter/${chapterIndex}`))
+        .data;
+
+    const isSameChapter = nextChapter.chapterNumber === currentChapter.chapterNumber;
+    const isSameScanlator = nextChapter.scanlator === currentChapter.scanlator;
+
+    return isSameChapter && !isSameScanlator;
+};
+
+const getPrevNextChapter = async (
+    chapterIndex: number,
+    currentChapter: IChapter,
+    skipDupChapters: boolean,
+    getNextChapterIndex: (skippedChapterIndex: number) => number,
+): Promise<number> => {
+    if (!skipDupChapters) {
+        return chapterIndex;
+    }
+
+    if (await shouldSkipChapter(chapterIndex, currentChapter)) {
+        return getPrevNextChapter(
+            getNextChapterIndex(chapterIndex),
+            currentChapter,
+            skipDupChapters,
+            getNextChapterIndex,
+        );
+    }
+
+    return chapterIndex;
+};
+
+export const getPrevChapter = async (currentChapter: IChapter, skipDupChapters: boolean): Promise<number> =>
+    getPrevNextChapter(currentChapter.index - 1, currentChapter, skipDupChapters, (skippedIndex) => skippedIndex - 1);
+
+export const getNextChapter = async (currentChapter: IChapter, skipDupChapters: boolean): Promise<number> =>
```

Comment:
This and `getPrevChapter` could be refactored into single function, which takes `offset` parameter. It would be +1 or -1, which could be then used to get the chapter index, and it could be passed further down to `getPrevNextChapter` instead of the last function.

In the end, there can only be `getOffsetChapter` which takes the offset number. I presume the behavior is not different between going forward or backward.

Reply:
not sure why I did it that way in the first place